src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6613,6 +6613,19 @@ private StatementSyntax ParsePossibleDeclarationOrBadAwaitStatement(ref ResetPoi
         /// </remarks>
         private StatementSyntax ParseStatementNoDeclaration(bool allowAnyExpression)
         {
+            if (this.CurrentToken.ContextualKind == SyntaxKind.AwaitKeyword)
+            {
+                switch (this.PeekToken(1).Kind)
+                {
+                    case SyntaxKind.UsingKeyword:
+                        return this.ParseUsingStatement(ParseAwaitKeywordForAsyncStreams());
+                    case SyntaxKind.ForEachKeyword:
+                        return this.ParseForEachStatement(ParseAwaitKeywordForAsyncStreams());
```

Comment:
the name of this method seems wonky (given that it is used for 'using statements' as well).

Reply:
"No declaration" seems short for "no local declaration statement". So didn't seem that bad.

Reply:
>  It is more general than async enumerables or async loops.

enhhhhhhhhhhhhh.  still don't like hte name :)